src/java.base/share/classes/jdk/internal/value/ValueClass.java

```diff
@@ -27,17 +27,30 @@
 
 import jdk.internal.access.JavaLangReflectAccess;
 import jdk.internal.access.SharedSecrets;
+import jdk.internal.misc.PreviewFeatures;
 import jdk.internal.vm.annotation.IntrinsicCandidate;
 
-import java.lang.reflect.Array;
 import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
 
 /**
  * Utilities to access package private methods of java.lang.Class and related reflection classes.
  */
-public class ValueClass {
+public final class ValueClass {
     private static final JavaLangReflectAccess JLRA = SharedSecrets.getJavaLangReflectAccess();
 
+    /// {@return whether this field type may store value objects}
+    /// This excludes primitives and includes Object.
+    public static boolean isValueObjectCompatible(Class<?> fieldType) {
+        return PreviewFeatures.isEnabled() && !fieldType.isPrimitive() && (!fieldType.isIdentity() || fieldType == Object.class);
```

Comment:
The `fieldType == Object.class` would always be true regardless of enable-preview.
That does not seem to be accurate and might be mis-interpreted by a caller.